hazelcast/src/main/java/com/hazelcast/cache/impl/CachePartitionSegment.java

```diff
@@ -131,4 +133,14 @@ void clearHavingLesserBackupCountThan(int backupCount) {
             }
         }
     }
+
+   public Collection<ServiceNamespace> getAllNamespaces(int replicaIndex) {
+        Collection<ServiceNamespace> namespaces = new HashSet<ServiceNamespace>();
```

Comment:
As far as I know, no, each record-store will have a different name. We can use a list too.